app/src/main/java/com/matmoongi/viewmodels/SearchViewModel.kt

```diff
@@ -68,6 +91,9 @@ class SearchViewModel(
         TODO("네아로로 로그인 상태 확인")
     }
 
+    fun loginWithNaver(context: Context, oAuthLoginCallback: OAuthLoginCallback): () -> Unit =
+        { NaverIdLoginSDK.authenticate(context, oAuthLoginCallback) }
```

Comment:
줄바꿈이 어색해서 읽기가 약간 버겁습니다.
```suggestion
    fun loginWithNaver(
        context: Context,
        oAuthLoginCallback: OAuthLoginCallback
    ): () -> Unit = {
        NaverIdLoginSDK.authenticate(context, oAuthLoginCallback)
    }
```

줄바꿈을 정리하고 보니, 재미있는 사실을 알았습니다.
`loginWithNaver()` 함수이길래, 이 함수를 호출하면 로그인을 시작하는가 싶었는데 그것이 아니군요.

이름은 `loginWithNaver()` 이지만, 로그인을 시작하는 함수를 생성해서 반환하는 함수이군요?

Reply:
네 생각해보니 onClickLoginButton이 더 적절할 것 같습니다.